src/DefaultBuilder/src/WebApplicationBuilder.cs

```diff
@@ -28,20 +28,23 @@ internal WebApplicationBuilder(Assembly? callingAssembly, string[]? args = null)
             // HACK: MVC and Identity do this horrible thing to get the hosting environment as an instance
             // from the service collection before it is built. That needs to be fixed...
             Environment = _environment = new WebHostEnvironment(callingAssembly);
+
             Services.AddSingleton(Environment);
 
             // Run methods to configure both generic and web host defaults early to populate config from appsettings.json
             // environment variables (both DOTNET_ and ASPNETCORE_ prefixed) and other possible default sources to prepopulate
             // the correct defaults.
             var bootstrapBuilder = new BootstrapHostBuilder(Configuration, _environment);
-            bootstrapBuilder.ConfigureDefaults(args);
             bootstrapBuilder.ConfigureWebHostDefaults(configure: _ => { });
+            bootstrapBuilder.ConfigureDefaults(args);
```

Comment:
Flipping the order of configuring WebHostDefaults was important for ensuring that the right appsettings.{Env}.json gets loaded initially when `ASPNETCORE_ENVIRONMENT` is configured. I don't know of any negative consequences of flipping the default configuration order for the BootstrapHostBuilder, but that doesn't mean there aren't any.